### PR TITLE
feat(state_table): add pk_indices in state_table

### PR DESCRIPTION
**Pull request by wcy-fdu**
## What's changed and what's your intention?
As title.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/storage/src/table/state_table.rs [hunk 1]
```diff
@@ -45,13 +45,16 @@ pub struct StateTable<S: StateStore> {
...
+
+    _pk_index: Vec<usize>,
 }
 impl<S: StateStore> StateTable<S> {
     pub fn new(
         keyspace: Keyspace<S>,
         column_descs: Vec<ColumnDesc>,
         order_types: Vec<OrderType>,
         dist_key_indices: Option<Vec<usize>>,
+        _pk_index: Vec<usize>,
```
Comment (fuyufjh) [inline src/storage/src/table/state_table.rs:57]: Why started with an underscore (`_`) ?

Comment (fuyufjh) [reply src/storage/src/table/state_table.rs:57]: By the way, recommend `pk_indices`

**Review by fuyufjh — COMMENTED** [review]
Rest LGTM

File: src/storage/src/table/state_table.rs (hunk 1 above)
Comment (fuyufjh) [reply src/storage/src/table/state_table.rs:57]: By the way *2, `OrderPair` looks better than `order_types & pk_index` ?

Comment (wcy-fdu) [reply src/storage/src/table/state_table.rs:57]: It hasn't been used yet, and will be used in top_n_state.

**Review by fuyufjh — APPROVED** [review]

**Pull request merged**